app/src/main/java/com/example/parkingsystemkotlin/mvp/model/ReservationModel.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.parkingsystemkotlin.mvp.model
+
+import com.example.parkingsystemkotlin.mvp.contract.ReservationContract
+import com.example.parkingsystemkotlin.mvp.model.reservation.DataBase
+import com.example.parkingsystemkotlin.mvp.model.reservation.Reservation
+import java.util.Calendar
+
+class ReservationModel(private val reservationInformationDB: DataBase) : ReservationContract.ReservationModelContract {
+
+    private var isStartDateAndTime: Boolean = true
+    private lateinit var startDateAndTime: Calendar
+    private lateinit var finishDateAndTime: Calendar
+
+    override fun saveDate(year: Int, month: Int, dayOfMonth: Int) {
+        if (isStartDateAndTime) {
+            startDateAndTime = Calendar.getInstance()
+            startDateAndTime.set(Calendar.DAY_OF_MONTH, dayOfMonth)
+            startDateAndTime.set(Calendar.MONTH, month - ONE)
+            startDateAndTime.set(Calendar.YEAR, year)
+        } else {
+            finishDateAndTime = Calendar.getInstance()
+            finishDateAndTime.set(Calendar.DAY_OF_MONTH, dayOfMonth)
+            finishDateAndTime.set(Calendar.MONTH, month - ONE)
+            finishDateAndTime.set(Calendar.YEAR, year)
+        }
+    }
+
+    override fun saveTime(hourOfDay: Int, minute: Int) {
+        if (isStartDateAndTime) {
+            startDateAndTime.set(Calendar.HOUR_OF_DAY, hourOfDay)
+            startDateAndTime.set(Calendar.MINUTE, minute)
+        } else {
+            finishDateAndTime.set(Calendar.HOUR_OF_DAY, hourOfDay)
+            finishDateAndTime.set(Calendar.MINUTE, minute)
+        }
+    }
+
+    override fun saveReservation(securityCode: String, place: String) {
+        if (::startDateAndTime.isInitialized && ::startDateAndTime.isInitialized && !securityCode.isEmpty() && !place.isEmpty()) {
+            val reservation = Reservation(startDateAndTime, finishDateAndTime, securityCode, place)
```

Comment:
if you are using this var just one time you can create this on putReservationDB

Reply:
fixed